common/src/main/java/io/netty/util/Recycler.java

```diff
@@ -232,19 +232,42 @@ public void recycle(Object object) {
                 next = stack.head;
                 stack.head = this;
             }
-            this.stack = stack;
+
+            // Its important that we not store the Stack itself in the WeakOrderQueue as the Stack also is used in
+            // the WeakHashMap as key. So just store the enclosed AtomicInteger which should allow to have the
+            // Stack itself GCed.
+            availableSharedCapacity = stack.availableSharedCapacity;
```

Comment:
the stack doesn't actually use this variable ... should we move it outside the stack and lazy create it here in the `WeakOrderQueue`?